### DISCUSSION TITLE
Is there a way to generate d.ts files during dev?

### DISCUSSION THREAD

**Discussion by theogravity** [alex8088/electron-vite#102] [Q&A]
I have my `tsconfig.node.json` with the following definition:

```
{
  "extends": "@electron-toolkit/tsconfig/tsconfig.node.json",
  "include": ["electron.vite.config.*", "src/**/*"],
  "compilerOptions": {
    "composite": true,
    "declaration": true,
    "declarationDir": "src/types",
    "types": ["electron-vite/node"],
    "target": "ES2022",
    "module": "esnext",
    "moduleResolution": "node"
  },
}
```

When I run `electron-vite dev --watch`, I was hoping that the `d.ts` files would be generated on reload, but this doesn't happen.

The reason why I want to do this is so I can share the types directly with `src/renderer` in some cases without referencing `src/main` directly. I like to keep my type and interface definitions with the code it's tied to, but those files cannot be imported into the renderer for technical reasons since they contain non-type exports.

**Comment by alex8088** [comment]
You can use `tsc`.

**Reply by theogravity** [reply]
I was wondering if it could be executed as part of the watch so the `d.ts` file would re-generate on file changes.

**Reply by alex8088** [reply]
Currently there is no hook to do this.
